app/common/model/src/test/java/io/syndesis/common/model/DataShapeBuilderTest.java

```diff
@@ -0,0 +1,210 @@
+                                .build())
+                .addVariant(new DataShape.Builder()
+                                .specification(variantSpecification)
+                                .build());
+
+        Assert.assertEquals(specification, shape.build().getSpecification());
+        Assert.assertEquals(specification, shape.compress().build().getSpecification());
+        Assert.assertEquals(specification, shape.compress().decompress().build().getSpecification());
+        Assert.assertEquals(variantSpecification, shape.build().getVariants().get(0).getSpecification());
+        assertCompressedSpecification(variantSpecification, shape.compress().build().getVariants().get(0).getSpecification());
+        Assert.assertEquals(variantSpecification, shape.build().getVariants().get(1).getSpecification());
+        Assert.assertEquals(variantSpecification, shape.compress().build().getVariants().get(1).getSpecification());
+    }
+
+    @Test
+    public void shouldNotCompressAlreadyCompressedVariants() throws IOException {
+        DataShape.Builder shape = new DataShape.Builder()
+                .putMetadata(DataShape.Builder.COMPRESSION_METADATA_KEY, "true")
+                .specification(specification)
+                .addVariant(new DataShape.Builder()
+                        .putMetadata(DataShape.Builder.COMPRESSION_METADATA_KEY, "true")
+                        .specification(variantSpecification)
+                        .compress()
+                        .build());
+
+        assertCompressedSpecification(specification, shape.compress().compress().build().getSpecification());
+        assertCompressedSpecification(variantSpecification, shape.compress().compress().build().getVariants().get(0).getSpecification());
+    }
+
+    @Test
+    public void shouldNotDecompressAlreadyDecompressedVariants() {
+        DataShape.Builder shape = new DataShape.Builder()
+                .putMetadata(DataShape.Builder.COMPRESSION_METADATA_KEY, "true")
+                .specification(compressedSpecification)
+                .addVariant(new DataShape.Builder()
+                        .putMetadata(DataShape.Builder.COMPRESSION_METADATA_KEY, "true")
+                        .specification(variantSpecification)
+                        .compress()
+                        .build());
+
+        Assert.assertEquals(specification, shape.decompress().decompress().build().getSpecification());
+        Assert.assertEquals(variantSpecification, shape.decompress().decompress().build().getVariants().get(0).getSpecification());
+    }
+
+    private void assertCompressedSpecification(String expected, String compressed) throws IOException {
+        byte[] data = Base64.getDecoder().decode(compressed);
+
+        try (InputStream is = new GZIPInputStream(new ByteArrayInputStream(data))) {
+            Assert.assertEquals(expected, readToString(is));
+        }
+    }
+
+    private String readToString(InputStream is) throws IOException {
```

Comment:
`io.syndesis.common.util.IOStreams::readText`

Reply:
Ah nice. Haven't seen this one. 

I have had `org.apache.commons.io.IOUtils` before but this introduces a new compile time dependency to `syndesis-model` and to all modules that depend on `syndesis-model`.

The common util helper in Syndesis matches perfectly, thanks!